metricq_source_bacnet/source.py

```diff
@@ -418,6 +426,21 @@ async def _worker_task(self, object_group, worker_task_stop_future):
                 device_address_str, objects, chunk_size=chunk_size
             )
 
+            if object_group.get("nan_at_timeout"):
+
+                for metric_id in metrics:
+                    now = Timestamp.now()
+                    last_timestamp = self._last_value_send_by_metric.get(metric_id, now)
+                    if now - last_timestamp >= 6 * Timedelta.from_s(interval):
+                        timestamp_nan = last_timestamp + 5 * Timedelta.from_s(interval)
```

Comment:
I think you're placing the NaN at the wrong time. What if, for instance `now - last_timestamp == 10 * interval`? Then the NaN will be written with at `now - 5 * interval` and there will be a `5 * interval` gap, right?

```suggestion
                        timestamp_nan = now - Timedelta.from_s(interval)
```

Reply:
Yes, and in the next loop the condition is met again and a NaN is written with `now - interval`